docs/reference/mapping/types.asciidoc

```diff
@@ -1,72 +1,115 @@
 [[mapping-types]]
 == Field data types
 
-Elasticsearch supports a number of different data types for the fields in a
-document:
+Each field has a _field data type_, or _field type_. This type indicates the
+kind of data the field contains, such as strings or boolean values, and its
+intended use. For example, you can index strings to both `text` and `keyword`
+fields. However, `text` field values are <<analysis,analyzed>> for full-text
+search while `keyword` strings are left as-is for filtering and sorting.
+
+Field types are grouped by _family_. Types in the same family support the same
+search functionality but may have different space usage or performance
+characteristics.
+
+Currently, the only type family is `keyword`, which consists of the `keyword`,
+`constant_keyword`, and `wildcard` field types. Other type families have only a
+single field type. For example, the `boolean` type family consists of one field
+type: `boolean`.
+
 
 [discrete]
 [[_core_datatypes]]
-=== Core data types
+==== Common types
+
+<<binary,`binary`>>::   Binary value encoded as a Base64 string.
+<<boolean,`boolean`>>:: `true` and `false` values.
+Keyword::               The keyword family, including <<keyword, `keyword`>>,
+                        <<constant-keyword,`constant_keyword`>>, and
+                        <<wildcard, `wildcard`>>.
+<<number,Numbers>>::    Numeric types, such as `long` and `double`, used to
```

Comment:
Sorry for nitpicking -- the 'amounts' part is really helpful, but the 'rather than IDs' part feels confusing, and as @markharwood mentioned there are actually some use cases for mapping ordered IDs as numbers. We could remove 'rather than IDs' (since we already have a warning/ tip about it in the numeric types docs)?

Reply:
No problem. Removed with c4beb426f13. Thanks for raising.